tests/test_test.py

```diff
@@ -0,0 +1,428 @@
+        result = self.run_test(
+            "-v", "0", f"{__name__}.ExampleTests.test_unexpected_success"
+        )
+        assert result.returncode == 0
+        assert result.stderr.splitlines()[:1] == [
+            "-" * 70,
+        ]
+
+    def test_unexpected_success_normal(self):
+        result = self.run_test(f"{__name__}.ExampleTests.test_unexpected_success")
+        assert result.returncode == 0
+        assert result.stderr.splitlines()[:2] == [
+            "u",
+            "-" * 70,
+        ]
+
+    def test_unexpected_success_verbose(self):
+        result = self.run_test(
+            "-v", "2", f"{__name__}.ExampleTests.test_unexpected_success"
+        )
+        assert result.returncode == 0
+        assert result.stderr.splitlines()[:3] == [
+            (
+                "test_unexpected_success (tests.test_test.ExampleTests) ... "
+                + "unexpected success"
+            ),
+            "",
+            "-" * 70,
+        ]
+
+    # Need to refactor RichDebugSQLTextTestResult at least to add printErrorList
+    # def test_debug_sql(self):
+    #     stderr = subprocess.run(
+    #         [
+    #             "python",
+    #             "-m",
+    #             "django",
+    #             "test",
+    #             "--debug-sql",
+    #             f"{__name__}.ExampleTests.test_assertion",
+    #         ],
+    #         capture_output=True,
+    #         text=True,
+    #     ).stderr
+    #     assert "─ locals ─" in stderr
+    #     assert "Tests can have descriptions." in stderr
+
+    @pytest.mark.skipif(django.VERSION < (3,), reason="--pdb added in Django 3.0")
+    def test_pdb(self):
+        result = self.run_test(
+            "--pdb",
+            f"{__name__}.ExampleTests.test_failure",
+            input="p 2 * 2\n",
+        )
+
+        assert result.returncode == 1
+        lines = result.stdout.splitlines()
+        expected = [
+            "System check identified no issues (0 silenced).",
+            "",
+            "Opening PDB: AssertionError('1 != 2')",
+        ]
+        if django.VERSION >= (4,):
+            expected.insert(0, "Found 1 test(s).")
+        assert lines[: len(expected)] == expected
+
+        assert lines[len(expected) + 1 : len(expected) + 4] == [
+            "-> raise self.failureException(msg)",
+            "(Pdb) 4",
+            "(Pdb) ",
+        ]
+
+    def test_failure_django_assertion(self):
+        "Django and Unit test modules are hidden in tracebacks."
+        result = self.run_test(f"{__name__}.ExampleTests.test_failure_django_assertion")
+        assert result.returncode == 1
+        assert "testPartExecutor" not in result.stderr
+        assert "_callTestMethod" not in result.stderr
+        assert 'self.assertURLEqual("/url/", "/test/")' in result.stderr
+        assert 'self.assertURLEqual("/url/", "/test/")' in result.stderr
+        assert result.stderr.count("─ locals ─") == 1
+
+
+class TestDebugSQL(unittest.TestCase):
```

Comment:
Yeah, I undid this. I don't think we need to check every case, and we can do so more simply with simple queries like `SELECT 1234`, rather than using models.